h/api/models.py

```diff
@@ -166,6 +176,76 @@ def title(self):
             return ""
 
     @property
+    def filename(self):
+        if self.uri and self.uri.startswith("file://"):
+            return self.uri.split("/")[-1] or None
+
+    @property
+    def document_link(self):
+        """Return a link to this annotation's document.
```

Comment:
It seems odd to try and render the HTML outside of the template. You could just always render the `<a>` tag in the markup and only include the `href` attribute if a URL is available.